include/seqan3/range/decorator/gap_decorator_anchor_set.hpp

```diff
@@ -0,0 +1,767 @@
+                    {
+                        left_gap_end = 0;
+                    }
+                }
+            }
+            else if (anchor_set_it == (*host).anchors.end() || pos < (*anchor_set_it).first)
+            {   // we are neither at the left nor right gap
+                --ungapped_view_pos;
+            }
+            // else -> no op (we are still within the right gap stretch)
+
+            return *this;
+        }
+
+        //!\brief Post-increment, returns previous iterator state (delegates to pre-increment).
+        constexpr gap_decorator_anchor_set_iterator operator++(int) noexcept
+        {
+            gap_decorator_anchor_set_iterator cpy{*this};
+            ++(*this);
+            return cpy;
+        }
+
+        //!\brief Post-decrement, returns previous iterator state (delegates to pre-decrement).
+        constexpr gap_decorator_anchor_set_iterator operator--(int) noexcept
+        {
+            gap_decorator_anchor_set_iterator cpy{*this};
+            --(*this);
+            return cpy;
+        }
+        //!\}
+
+        /*!\name Reference/Dereference operators
+         * \{
+        */
+        //!\brief Dereference operator returns element currently pointed at.
+        constexpr reference operator*() const noexcept
+        {
+            if (pos >= left_gap_end && (anchor_set_it == (*host).anchors.end() || pos < (*anchor_set_it).first))
+                return value_type{(*host).ungapped_view[ungapped_view_pos]};
+            else
+                return gap{};
+        }
+
+        //!\brief Returns pointer to this iterator.
+        constexpr pointer operator->() const noexcept
+        {
+            // ?
+        }
+        //!\}
+
+        /*!\name Comparison operators
+         * \brief Compares iterators by virtual position.
+         * \{
+         */
+        constexpr bool operator==(gap_decorator_anchor_set_iterator const & rhs) const noexcept
+        {
+            return pos == rhs.pos;
+        }
+
+        constexpr bool operator!=(gap_decorator_anchor_set_iterator const & rhs) const noexcept
+        {
+            return pos != rhs.pos;
+        }
+
+        constexpr bool operator<(gap_decorator_anchor_set_iterator const & rhs) const noexcept
+        {
+            return static_cast<bool>(pos < rhs.pos);
+        }
+
+        constexpr bool operator>(gap_decorator_anchor_set_iterator const & rhs) const noexcept
+        {
+            return pos > rhs.pos;
+        }
+
+        constexpr bool operator<=(gap_decorator_anchor_set_iterator const & rhs) const noexcept
+        {
+            return pos <= rhs.pos;
+        }
+
+        constexpr bool operator>=(gap_decorator_anchor_set_iterator const & rhs) const noexcept
+        {
+            return pos >= rhs.pos;
+        }
+        //!\}
+    };
+
+public:
+    /*!\name Member types
+     * \{
+     */
+    //!\brief The underlying ungapped range type.
+    using ungapped_range_type = inner_type;
```

Comment:
I would not group this with the others because it's not one of the typically "range-associated" types.